mapbox_vector_tile/polygon.py

```diff
@@ -205,9 +209,39 @@ def make_it_valid(shape):
         return shape
 
     elif shape.type == 'MultiPolygon':
-        shape = make_valid_multipolygon(shape)
+        shape = make_valid_multipolygon(shape, asserted)
 
     elif shape.type == 'Polygon':
-        shape = make_valid_polygon(shape)
+        shape = make_valid_polygon(shape, asserted)
 
     return shape
+
+
+def clean_multi(shape):
+    """
+    Remove self- and ring-selfintersections from input Polygon geometries
+    """
+    polygons = []
+    for p in polygons:
```

Comment:
I can't see where the `shape` parameter is used. Was that supposed to be `for p in shape.geoms`?